src/Navigation/Plugins/recastJSPlugin.ts

```diff
@@ -0,0 +1,267 @@
+import { INavigationEnginePlugin, ICrowd, AgentParameters, NavMeshParameters } from "../../Navigation/INavigationEngine";
+import { Logger } from "../../Misc/logger";
+import { VertexData } from "../../Meshes/mesh.vertexData";
+import { AbstractMesh } from "../../Meshes/abstractMesh";
+import { Mesh } from "../../Meshes/mesh";
+import { Scene } from "../../scene";
+import { Vector3 } from '../../Maths/math';
+import { TransformNode } from "../../Meshes/transformNode";
+
+declare var Recast: any;
+
+/**
+ * RecastJS navigation plugin
+ */
+export class RecastJSPlugin implements INavigationEnginePlugin {
+    /**
+     * Reference to the Recast library
+     */
+    public bjsRECAST: any = {};
+    public name: string = "RecastJSPlugin";
+    public navMesh: any;
+
+    /**
+     * Initializes the recastJS plugin
+     */
+    public constructor(recastInjection: any = Recast) {
+        if (typeof recastInjection === "function") {
+            recastInjection(this.bjsRECAST);
+        } else {
+            this.bjsRECAST = recastInjection();
+        }
+
+        if (!this.isSupported()) {
+            Logger.Error("RecastJS is not available. Please make sure you included the js file.");
+            return;
+        }
+    }
+
+    /**
+     * Creates a navigation mesh
+     * @param mesh of all the geometry used to compute the navigatio mesh
+     * @param parameters bunch of parameters used to filter geometry
+     */
+    createMavMesh(mesh: AbstractMesh, parameters: NavMeshParameters): void {
+        var rc = new this.bjsRECAST.rcConfig();

```

Comment:
Use `const` if possible or use `let` otherwise. Avoid using `var` as it has scoping issues.